src/components/common/Input.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { ChangeEvent, InputHTMLAttributes } from 'react';
+import styled from '@emotion/styled';
+import { css } from '@emotion/react';
+
+type Sort = 'left' | 'middle';
+
+interface InputProps extends Omit<InputHTMLAttributes<HTMLInputElement>, 'maxLength'> {
+  className?: string;
+  sort: Sort;
+  valid?: boolean;
+  maxLength?: number;
+  [key: string]: any;
+}
+function Input({ className, sort, valid = true, maxLength, onChange, ...props }: InputProps) {
+  const handleChangeWithMaxLength = (e: ChangeEvent<HTMLInputElement>) => {
+    if (String(e.target.value).length <= maxLength && typeof e.target.value === 'string') {
+      onChange(e);
+    }
+  };
```

Comment:
e.target.value는 무조건 string type 이지 않나요?
type=number에서 number type으로 뽑아오는 것은 `e.target.valueAsNumber` 입니당.

아래처럼 하면 동작 안 하나요?

```suggestion
  const handleChangeWithMaxLength = (e: ChangeEvent<HTMLInputElement>) => {
    if (e.target.value <= maxLength) {
      onChange(e);
    }
  };
```

Reply:
```suggestion
  const handleChangeWithMaxLength = (e: ChangeEvent<HTMLInputElement>) => {
    if (String(e.target.value).length <= maxLength && typeof e.target.valueAsNumber === 'number') {
      onChange(e);
    }
  };
```
이렇게 하니까 정확히 원하는대로 동작합니다 !  e.target.valueAsNumber 배워갑니다 ㅎㅎ

Reply:
`String(e.target.value).length <= maxLength`
e.target.value는 무조건 string인데, String 형변환이 필요한가요?

`typeof e.target.valueAsNumber === 'number'`
e.target.valueAsNumber는 무조건 number를 return하는데, number 체크가 필요한가요?

Reply:
string형변환은 필요 없을 것 같습니다!
`typeof e.target.valueAsNumber === 'number'`
이 조건을 넣은 이유는,  Input에 숫자가 아니라 문자가 입력될 수도 있기 때문입니다 !
그래서 숫자가 입력되었을 때에만 onChange를 걸어서 입력될 수 있도록 저 조건을 넣었습니다 
(저 조건을 추가하지 않으면 일단 문자 하나는 인풋에 입력이 되고 나서 입력이 막힙니다)